### ISSUE TITLE
PostgreSQL.tt Generates All Database Objects

### ISSUE THREAD

**Issue by tekzoom** [linq2db/linq2db#1919]
The PostgreSQL.tt generates all database objects including catalogs, functions, all schema, etc. How to just generate models of public schema?
Thanks for looking into this. tz

**Comment by MaceWindu** [comment]
Check available options [here](GetSchemaOptions.IncludedSchemas)

You need something like this:
```cs
// to inlcude only public. members
GetSchemaOptions.IncludedSchemas = new[] { "public" };
// to disable functions
GetSchemaOptions.GetProcedures = false;
```

**Comment by tekzoom** [comment]
Thanks for your reply. I tried the suggested options above and others from the options list but no luck. I still get a generated file with all catalogs, functions, etc. This only happens for PostgreSQL. The MySQL and SQL Server work perfectly fine.

**Comment by MaceWindu** [comment]
Just rechecked it and it worked for me. Could it be that you put `GetSchemaOptions` settings after `LoadPostgreSQLMetadata`? `GetSchemaOptions` settings should be configured before this call, becuase they actually specify how it should work.

**Comment by MaceWindu** [comment]
I think we should add those notes directly to sample tt files to reduce confusion.

**Comment by tekzoom** [comment]
I added the GetSchemaOptions before the LoadPostgreSQLMetadata and it worked like a charm.

Thanks so much!

**Comment by MaceWindu** [comment]
closed with #1929
